### ISSUE TITLE
Mixture of experts Example

### ISSUE THREAD

**Issue by kiranmaya** [SciSharp/Keras.NET#260]
need MoE example, ChatGPT wrote some thing ,but ,I don't thinking its a good gating network 
`using System;
using Tensorflow;
using static Tensorflow.Binding;
using Tensorflow.Keras;
using Tensorflow.Keras.Layers;
using Tensorflow.Keras.Models;

class Program
{
    static void Main(string[] args)
    {
        int inputDim = 10;
        int outputDim = 1;
        int numExperts = 3;

        // Create experts
        var experts = new Sequential[numExperts];
        for (int i = 0; i < numExperts; i++)
        {
            experts[i] = CreateExpert(inputDim, outputDim);
        }

        // Input tensor
        var inputLayer = keras.Input(shape: new int[] { inputDim });
        
        // Output from dynamic routing
        var output = DynamicRouting(inputLayer, experts, numExperts);
        
        // Build model
        var moeModel = keras.Model(inputLayer, output);
        moeModel.compile(optimizer: keras.optimizers.Adam(), loss: "mse");

        moeModel.summary(); // Optional: to view the model summary
    }

    static Sequential CreateExpert(int inputDim, int outputDim)
    {
        var model = keras.Sequential();
        model.add(keras.Input(shape: new int[] { inputDim }));
        model.add(new Dense(64, activation: keras.activations.Relu));
        model.add(new Dense(outputDim, activation: keras.activations.Linear));
        return model;
    }

    static Tensor DynamicRouting(Tensor inputTensor, Sequential[] experts, int numExperts, int numActions = 2)
    {
        // Policy network to select experts dynamically
        var policyLayer = new Dense(numActions, activation: keras.activations.Softmax).Apply(inputTensor);
        
        // Expert outputs
        var expertOutputs = new Tensor[numExperts];
        for (int i = 0; i < numExperts; i++)
        {
            expertOutputs[i] = experts[i].Apply(inputTensor);
        }

        // Selecting the expert with the highest probability
        var selectedExpert = tf.argmax(policyLayer, axis: 1);
        
        // Gather the output from the selected expert
        var output = tf.gather(expertOutputs, selectedExpert);
        
        return output;
    }
}
`
